### PR TITLE
Create temporal auto-setup docker and pipeline

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Adds dockerfile and pipeline that builds temporaliotest/auto-setup docker image on each commit and pushes to Docker Hub

## Why?

automation

## Checklist

1. Closes 

2. How was this tested:

Ran a setup using https://github.com/temporalio/docker-compose/blob/main/docker-compose-ui-experimental.yml and replacing auto-setup image with the image built from the new auto-setup.Dockerfile

3. Any docs updates needed?

### CODE REVIEW

File: .github/workflows/docker-admin-tools.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Publish to Docker Hub
+name: Publish Adming Tools image
```
Comment (underrun) [inline .github/workflows/docker-admin-tools.yml:1]: Adming -> Admin

File: auto-setup.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
+ARG BASE_BUILDER_IMAGE=temporalio/base-builder:1.4.0
+ARG BASE_SERVER_IMAGE=temporalio/base-server:1.4.0
```
Comment (yiminc) [inline auto-setup.Dockerfile:2]: server is upgrading to 1.5.0: https://github.com/temporalio/temporal/pull/2402

File: auto-setup.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+COPY ./tctl/go.mod ./tctl/go.sum ./tctl/
+RUN (cd ./tctl && go mod download)
+
+# build
+COPY . .
+RUN (cd ./temporal && make bins)
+RUN (cd ./tctl && make build)
+
+##### Temporal server #####
+FROM ${BASE_SERVER_IMAGE} as temporal-server
```
Comment (alexshtin) [inline auto-setup.Dockerfile:24]: If you are creating dedicated `Dockerfile` for auto-setup, you don't need this intermediate layer here.

File: auto-setup.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+COPY ./temporal/go.mod ./temporal/go.sum ./temporal/
+RUN (cd ./temporal && go mod download)
+
+# cache tctl packages as a docker layer
+COPY ./tctl/go.mod ./tctl/go.sum ./tctl/
+RUN (cd ./tctl && go mod download)
+
+# build
+COPY . .
+RUN (cd ./temporal && make bins)
```
Comment (alexshtin) [inline auto-setup.Dockerfile:20]:
```suggestion
RUN (cd ./temporal && CGO_ENABLED=0 make temporal-server)
```

File: auto-setup.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+RUN mkdir /etc/temporal/config
+RUN chown -R temporal:temporal /etc/temporal/config
+USER temporal
+
+COPY --from=temporal-builder /home/builder/tctl/tctl /usr/local/bin
+COPY --from=temporal-builder /home/builder/tctl/tctl-authorization-plugin /usr/local/bin
+COPY --from=temporal-builder /home/builder/temporal/temporal-server /usr/local/bin
+
+COPY temporal/config/dynamicconfig /etc/temporal/config/dynamicconfig
+COPY temporal/docker/config_template.yaml /etc/temporal/config/config_template.yaml
```
Comment (alexshtin) [inline auto-setup.Dockerfile:43]: May be in next PR, but if you move `Dockerfile`s to this repo,`temporal/docker` dir needs to be moved to this repo too.

**Review by alexshtin — CHANGES_REQUESTED** [review]

File: auto-setup.Dockerfile (hunk 3 above)
Comment (feedmeapples) [reply auto-setup.Dockerfile:24]: done

File: auto-setup.Dockerfile [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+FROM temporal-auto-setup as temporal-develop
+
+# apk and setup-develop.sh requires root permissions.
+USER root
+# iproute2 contains tc, which can be used for traffic shaping in resiliancy testing. 
+ONBUILD RUN apk add iproute2
+
+CMD ["autosetup", "develop"]
+
+COPY temporal/docker/setup-develop.sh /etc/temporal/setup-develop.sh
```
[the remark this answers is not in the document]
Comment (feedmeapples) [reply auto-setup.Dockerfile]: done

File: auto-setup.Dockerfile [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+# auto-setup files
+COPY --from=temporal-builder /home/builder/temporal/schema /etc/temporal/schema
+COPY --from=temporal-builder /home/builder/temporal/temporal-cassandra-tool /usr/local/bin
+COPY --from=temporal-builder /home/builder/temporal/temporal-sql-tool /usr/local/bin
+COPY temporal/docker/auto-setup.sh /etc/temporal/auto-setup.sh
+
+CMD ["autosetup"]
+
+USER temporal
```
Comment (alexshtin) [inline auto-setup.Dockerfile:57]: Dup of line 36.

File: auto-setup.Dockerfile [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+COPY temporal/config/dynamicconfig /etc/temporal/config/dynamicconfig
+COPY temporal/docker/config_template.yaml /etc/temporal/config/config_template.yaml
+COPY temporal/docker/entrypoint.sh /etc/temporal/entrypoint.sh
+COPY temporal/docker/start-temporal.sh /etc/temporal/start-temporal.sh
+
+# auto-setup files
+COPY --from=temporal-builder /home/builder/temporal/schema /etc/temporal/schema
+COPY --from=temporal-builder /home/builder/temporal/temporal-cassandra-tool /usr/local/bin
+COPY --from=temporal-builder /home/builder/temporal/temporal-sql-tool /usr/local/bin
+COPY temporal/docker/auto-setup.sh /etc/temporal/auto-setup.sh
```
Comment (alexshtin) [inline auto-setup.Dockerfile:53]: Please group all `COPY` by file type:
1. scripts
2. configs
3. binaries

File: develop.Dockerfile [hunk 9]
```diff
@@ -0,0 +1,70 @@
+ARG BASE_BUILDER_IMAGE=temporalio/base-builder:1.5.0
```
Comment (alexshtin) [inline develop.Dockerfile:1]: Same comments for this file.

File: auto-setup.Dockerfile (hunk 8 above)
Comment (feedmeapples) [reply auto-setup.Dockerfile:53]: done

**Review by alexshtin — APPROVED** [review]

**Pull request merged**